### ISSUE TITLE
Inline React components in MDX cause line lengths to exceed print-width

### ISSUE THREAD

**Issue by brandonweiss** [prettier/prettier#9503]
**Prettier 2.1.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeAogD3gJygQwBsAZASygGsACACxzgDMBeAHRBphgAckB6X-AHcAzgDp8AW3wAvaKMgTekACZwupLnELk4vNgD4AggHUAylQDCEVQAUNWnal5ZcBEuQr6qpYVXxVIWHIAVwhg31VtADc4HABPPyhlKnpCOHxhOD9gmAgpGFJoKkycKNIwLJgafBg-UmVfYQkICCqqVS5CCDiJBBgxKgBNMID8KG9YOABzHBqsgGYcZK58HBgE3IhCX3JcxITheC4qCAYqOLCcFK10zKouHAgK4V8IK4vg0fHwrKtbe20UCyGUSVAQygAtLkIeDilscoUoAAaCZgQjBZTkKZUADiNAghyoAEkoMJNGBcld4IdfGNksJgqRqaIQEiQBAuAVoMJkKBVo9BDZVggeSgiIJ8HEeWyAEazMAUOAwUwrMBY5AwHDBOBsuASGVwZSqZTEMZTYL4KZwABib3yBSgU2QIHwOQgrPYMAkhGMNCZcDJ+AqphFTNIZXWzrALw95BKMBssymUmQDCImTZACthJgAELyxXKyRwMhA1PpnUgbOYUxYtIARVC8HL20rKxwJWdEmUmA9DwJcGMsy4zoeAdiMQ9AEcm3BExzRS7hBCgYbDR76DPSPRE5aU0g0622ZkJKQNVrK8I63BG604C2MyAYPgZcZ6lVkAAmNma-CkQFTFYEj7iAAYAKwej8AAqL6ioej5RNqJKqLAphgDgGgwIYSSmOsaQPm2GGwG+ygfkgACMAAMlEAL40UAA)
```sh
--parser mdx
--print-width 100
```

**Input:**
```mdx
<ExternalLink href="http://aws.amazon.com/codepipeline/">AWS CodePipeline</ExternalLink> is a continuous delivery and release automation service that aids smooth deployments. You can integrate 3rd party tools into any step of your release process or you can use CodePipeline as an end-to-end solution, including Ghost Inspector tests and suites.
```

**Output:**
```mdx
<ExternalLink href="http://aws.amazon.com/codepipeline/">AWS CodePipeline</ExternalLink> is a continuous
delivery and release automation service that aids smooth deployments. You can integrate 3rd party tools
into any step of your release process or you can use CodePipeline as an end-to-end solution, including
Ghost Inspector tests and suites.

```

**Expected behavior:**

Here the output optically looks correct, however if you view it in the playground or any editor that has a line length “ruler” you can see that each line is actually longer than 100 characters.

```
❯ echo '<ExternalLink href="http://aws.amazon.com/codepipeline/">AWS CodePipeline</ExternalLink> is a continuous' | wc
       1       6     105
```

This causes a bit of a problem if you have word wrapping on in your editor.

<img width="1014" alt="Screen Shot 2020-10-26 at 1 58 54 PM" src="https://user-images.githubusercontent.com/4727/97228071-9c9b8000-1793-11eb-94c5-2669e9d7b01c.png">

If I remove the inline component it works fine, so the bug seems to have something to do with inline React components.

Thanks!

**Comment by thorn0** [comment]
Thanks for reporting this. If you're interested in fixing it, check the intermediate representation that Prettier builds by using the "show doc" checkbox in the playground. It's described [here](https://github.com/prettier/prettier/blob/master/commands.md). The `fill` command has [some expectations](https://github.com/prettier/prettier/blob/099c5fb7302021c894e38f53bb09909d45870502/src/document/doc-printer.js#L363) about its arguments, a violation of which may be the cause of this bug.

**Comment by brandonweiss** [comment]
I have a little bit of experience with ASTs/AST-like things so I took a look. I simplifed down the example and maybe noticed some things… it’s hard to tell because I’m not exactly what “normal” is supposed to look like.

I used two strings:

```
<em>hey</em>

<em>hey</em> is a continuous
```

<img width="568" alt="Screen Shot 2020-10-27 at 4 49 34 PM" src="https://user-images.githubusercontent.com/4727/97374390-67675e80-1875-11eb-9d5c-468e831aebec.png">

The HTML definitely seems to be parsed differently when there is markdown after it. It could be a concidence but those extra empty strings seem to match up to the amount of characters that the print width is exceeded by…
